File: code/android/app/src/main/java/at/avollmaier/tasknest/todo/data/CreateTodoDto.kt
package at.avollmaier.tasknest.todo.data

import java.time.ZonedDateTime

data class CreateTodoDto(
    val title: String,
    val content: String,
    val status: TodoStatus,
    val location: PointDto,
    val attachments: List<AttachmentDto> = emptyList(),
    val dueUtc: ZonedDateTime
)

File: code/backend/net.mstoegerer.TaskNest/net.mstoegerer.TaskNest.Api/Domain/DTOs/CreateTodoDto.cs
namespace net.mstoegerer.TaskNest.Api.Domain.DTOs;

public class CreateTodoDto
{
    public string Title { get; set; } = null!;
    public string Content { get; set; } = null!;
    public string Status { get; set; } = null!;
    public Guid? AssignedToId { get; set; }
    public DateTime? DueUtc { get; set; }
    public PointDto? Location { get; set; }
    public ICollection<CreateAttachmentDto> Attachments { get; set; } = new HashSet<CreateAttachmentDto>();
}